### ISSUE TITLE
CTC loss clarification

### ISSUE THREAD

**Issue by lbdave94** [ZhikangNiu/A-DMA#2]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

Hi, first of all thanks for sharing the code of your work. I have a question that about the CTC loss. In cfm_adma.py you define the CTC loss as 

```
ctc_loss += F.ctc_loss(
                z_tilde_ctc.transpose(1, 0).log_softmax(-1),  # Log probabilities (T, N, C)
                text,  # Target sequences (N*)
                z_lens_ctc,  # Length of inputs (N)
                text_lens,  # Length of targets (N)
                blank=self.transformer.text_embed.text_embed.num_embeddings,  # Blank token index
                reduction="mean",  # Reduction method ('none', 'mean', 'sum')
                zero_infinity=True,  # Ignore loss if log(0) happens
            )
```

why do you set blank like this instead of keep the default value 0 (or 1 according to vocab.txt)?

**Comment by arjun-gangwar** [comment]
Embedding at index 0 and index 1 are for the filler token and space token, respectively. The "blank" token in CTC is a separate token(Not to be confused with the "space" token). As far as, I could understand in code, they project the intermediate embeddings to `B*T_reduced*(num_embeddings+1)` which is your `z_tilde_ctc`.  So, when you calculate the log probabilities the final row for each `T_reduced*(num_embeddings+1)` matrix in `z_tilde_ctc` will correspond to the log probabilities of CTC blank token. I am not sure why they reduce the temporal dimension before calculating CTC loss. I am sure it would work without reducing temporal dimension as well. Maybe it gives them some computational benefits.
